src/convnets/convnext.jl

```diff
@@ -0,0 +1,91 @@
+"""
+    ConvNeXtBlock(planes, drop_path = 0.0f32, λ = 1f-6)
+
+Creates a single block of ConvNeXt.
+([reference](https://arxiv.org/pdf/2201.03545))
+
+# Arguments:
+- `planes`: number of input channels.
+- `drop_path`: Stochastic depth rate.
+- `λ`: Init value for [LayerScale](https://arxiv.org/pdf/2103.17239)
+"""
+function ConvNeXtBlock(planes, drop_path = 0.0f32, λ = 1f-6)
+  γ = λ > 0 ? Flux.ones32(planes) * λ : nothing
+  droppath = drop_path > 0 ? DropPath(drop_path) : identity
+  layers = SkipConnection(Chain(DepthwiseConv((7, 7), planes => planes, pad = 3), 
+                          x -> permutedims(x, (3, 1, 2, 4)),
+                          LayerNorm(planes; ϵ = 1f-6),
+                          mlpblock(planes, 4 * planes),
+                          x -> isnothing(γ) ? x : x .* γ,
+                          x -> permutedims(x, (2, 3, 1, 4)),
+                          droppath), +)
+  return layers
+end
+
+"""
+    convnext(; inchannels = 3, depths = [3, 3, 9, 3], planes = [96, 192, 384, 768], 
+               drop_path_rate = 0.0f32, λ = 1f-6, nclasses = 1000)
+
+Creates the layers for a ConvNeXt model.
+([reference](https://arxiv.org/pdf/2201.03545))
+
+# Arguments:
+- `inchannels`: number of input channels.
+- `depths`: list with configuration for depth of each block
+- `planes`: list with configuration for number of output channels in each block
+- `drop_path_rate`: Stochastic depth rate.
+- `λ`: Init value for [LayerScale](https://arxiv.org/pdf/2103.17239)
+- `nclasses`: number of output classes
+"""
+function convnext(; inchannels = 3, depths = [3, 3, 9, 3], planes = [96, 192, 384, 768], 
+                    drop_path_rate = 0.0f32, λ = 1fe-6, nclasses = 1000)
+  
+  layers = []
+  stem = Chain(Conv((4, 4), inchannels => planes[1]; stride = 4),
+               ChannelLayerNorm(planes[1]; ϵ = 1f-6))
+  push!(layers, stem)
+  dp_rates = LinRange{Float32}(0.0f32, drop_path_rate, sum(depths))
+  cur = 0
+  for i in 2:8
```

Comment:
Shouldn't this adjust with the length of `depths` and `planes`? I would recommend iterating `zip(depths, planes)`. The 2x iteration speed seems to be alternating between a downsampling layer and a sequence of blocks. Couldn't we just iterate at 1x and `push!` the downsampling layer then `append!` the blocks each iteration?

Reply:
That is what I was working with initially.....I don't remember why I changed it now 😂 I'll make it that way